src/System.Private.CoreLib/shared/System/Text/EncodingNLS.cs

```diff
@@ -56,7 +57,7 @@ public override unsafe int GetByteCount(char[] chars, int index, int count)
 
             // Just call the pointer version
             fixed (char* pChars = chars)
-                return GetByteCount(pChars + index, count, null);
+                return GetByteCount(pChars + index, count, null!); // TODO-NULLABLE: there is exactly one class overriding this and base class explicitly asserts this is non-nullable
```

Comment:
I'm not following; why is the `!` necessary?  If this is passing in `null`, doesn't that suggest that the parameter should be defined as nullable?

Reply:
Also, what's the "TODO" for?  i.e. what's the follow-up action you expect to need to do here?

Reply:
I believe one of the following things should happen:
- unify this class with the only derived class
- we should add explicit override of base which relaxes the non-null constraint (if possible make it abstract, if not make sure it always throws)
- alternatively change base class to be nullable (it's not exactly accurate since EncodingNLS is not the only class derived - couple of other deriving class also relax this though)

Some background:
We have couple of classes here:
- Encoding (base)
  - GetByteCount (and similar) which take non-nullable (with explicit Debug.Assert)
- EncodingNLS (derived - level 1)
  - assumes derived class will be overridden in such a way that GetByteCount will be nullable
- IIRC: Latin1Encoding (derived - level 2)
  - overrides GetByteCount and makes it nullable

Reply:
I'll add something similar to above in the TODO

Reply:
actually disregard, will just change base to be nullable - there doesn't seem to be much benefit in all these workaround considering this is internal method